packages/api/src/oauth/oauth.plugin.js

```diff
@@ -55,6 +56,19 @@ module.exports = {
       return Promise.all(cleanupModels.map(model => model.remove({ user }).exec()));
     });
 
+    server.method('isRedirectUriAllowedForApplication', (redirectUri, application) => {
```

Comment:
Wildcard is only at start of the domain name and you forgot to escape RegExp characters, so maybe you could write it like this:

```js
server.method('isRedirectUriAllowedForApplication', (redirectUri, application) => !!application.redirectUris.find(
  uri => new RegExp(uri.replace(/^\*/, '\\S+').replace(/[.*+?^${}()|[\]\\]/g, '\\$&')).test(redirectUri)
));
```